### ISSUE TITLE
Can't Suspend from Lock Screen

### ISSUE THREAD

**Issue by erenoglu** [linuxmint/Cinnamon#5919]
I am using a Surface Pro 3 on Arcy with Cinnamon 3.0.7. I use it with an external keyboard/mouse/screen both at work and at home. Here is the scenario. 
I work at night and let my computer on before going to sleep, so it can continue to download stuff, emails etc. The system automatically locks the screen after some time. 
In the morning, i unplug it from the usb hub, power cable, external monitor (i have others in the office). But I want to suspend it at this moment, since it'll go inside the laptop bag and no point for it to run.

Here is where I get stuck. At this state, I have the lock screen in front of me, I do not have the external keyboard connected (or the type cover), and the POwer Button do not respond to my suspend request. 
For it to suspend, I need to attach my Type Cover, type password, unlock, then press power button to suspend. 

It shall be possible to suspend from the lock screen in such case. This is the standard behavior as well on Windows 10.

**Comment by leigh123linux** [comment]
The lock screen is there to prevent unauthorized access, this includes preventing suspend without a password.
If you want to suspend it do it before disconnecting the keyboard!
As for the power button, that not working isn't a cinnamon issue (try filing an issue against the kernel instead).

**Comment by erenoglu** [comment]
Thanks for your comment. I think the "security" reason for not letting system suspend is not valid. At least it is not the case for no mainstream operating system. I am not worried for someone coming and suspending my system without my knowledge, I don't leave my computer anywhere unattended other than my office and home, and I'm not worried for security in neither of them. And if I do, I shall more be worried that getting my notebook stolen rather than just being suspended.

In my opinion, as a user, i shall not be bothered with logging in just to suspend, it is not intuitive. Ability to suspend from lock screen is offered by all Windows releases I've used since years and years. 

For power button, it is set to suspend and it is working properly when I'm logged in. It's just that this keyboard shortcut is omitted when I'm at lock screen.

**Comment by leigh123linux** [comment]
If you don't care about security you can disable lock screen in cinnamon-settings.

**Comment by erenoglu** [comment]
I do care about security, but not to the point of worrying about my system being suspended by a someone (as said, I never leave computer alone in untrusted place), but I still worry about colleagues staring at the screen or my kids deleting emails or sending garbage to others when I'm not there.
My request is simple, to have the option to have the suspend keys work at Lock Screen. Very common in all Windows systems. Gnome has it as well, but I like Cinnamon more.
(PS. I wonder what's the behavior when the lid is closed in lock screen state in classic notebooks)

**Comment by ManIVIctorious** [comment]
I also don't see any security violation in adding a suspend capability. If someone has physical access to my laptop he/she could as well shut down by pressing the power button for a while or pull the power plug.
Could you please elaborate in how suspending could be a security problem? (and please don't come with sidechannel attacks, every running system is vulnerable to those)

**Comment by erenoglu** [comment]
One more comment, actually, you can do a "switch user" and then choose the "suspend" from the menu in the new login screen. So it's possible, and no difference in terms of any security perception.
But doing this is not user friendly. So I'm looking forward to addition of the suspend related keyboard shortcuts to be enabled at lock screen state. (whenever it can be done!)

**Comment by chrisonhub** [comment]
@erenoglu 
did you know about the  gdmflexiserver command?  this is the equivalent to choosing "switch user" from the logout menu and will take you to the loginscreen - this has shutdown and suspend buttons on, right?

Maybe the answer is to call gdmflexiserver rather than cinnamon lock screen?  

Note that once your laptop is on the display manager's login screen, power settings are controlled by editing the etc/systemd/logind.conf file (at least on the 14.04 based Mint 17.x)

**Comment by erenoglu** [comment]
I'm not sure its gdmflexiserver. First of all I use lightdm, not gdm. And it the "switch user" in Cinnamon still works, so I wonder what's the alternative command. Checked from a terminal at switch user state but there's no sensible command there, lots of systemd and a line to start a new X on :1.

**Comment by chrisonhub** [comment]
@erenoglu yes, i think its an awkward work-around, but if the powers-that-be dont recognise this as a bug (I agree with you 100%, FWIW), then it might help.

According to the Arch wiki, there is a command-line control for lightdm, called dm-tool:
https://wiki.archlinux.org/index.php/LightDM#Command_line_tool

you could call this in place of the screensaver (i think) or make an applet or similar to run it

(its stuff like this which means we will never have a "linux year of the desktop") /flamewar

**Comment by lots0logs** [comment]
>you could call this in place of the screensaver (i think) or make an applet or similar to run it

You can use `light-locker` to use lightdm as your lock screen. The key to using it successfully is to make sure that it gets started before your session manager (so in this case before `cinnamon-session`). Easiest way to do that is to add this line at the end of `/etc/lightdm/Xsession` just before the call to start the session:
```
/usr/bin/light-locker &
sleep 3
```

**Comment by erenoglu** [comment]
Thanks Chrisonhub, Lots0logs. I tried "dm-tool lock" and it seems to do the job, it locks the current session and seems to switch to a new VT. Even if I try to go back to initial VT, it presents me the standard cinnamon lock screen. Unlocking the new VT also unlocks cinnamon screen lock. so seems good, but I'll need to find a way to replace the screenlock with this, without breaking the screenlock-on-current-VT function. Seems tough.
I also installed light-locker, and will test it.
